### ISSUE TITLE
More functiosns

### ISSUE THREAD

**Issue by DevilYangS** [MLanHu/GiftGame#2]
this is a fantastic code I have seen!!! Nice organization, perfect notes.
It is surprising for a rookie like me to understand the API.
Thank you for the contributors' work.

**Comment by MLanHu** [comment]
I agree with your words, but I still need to piont out the contribution is not an API. heng!
